settlement/src/tests.rs

```diff
@@ -0,0 +1,181 @@
+//! Various test scenarios for the settlement chain communication
+
+use super::*;
+use execution::*;
+use simperby_common::{verify::CommitSequenceVerifier, *};
+
+pub struct ChainInfo {
+    pub chain_name: String,
+    pub last_finalized_header: BlockHeader,
+    pub last_finalization_proof: FinalizationProof,
+    pub reserved_state: ReservedState,
+    /// The private keys of the validators of the next block.
+    ///
+    /// Both governance and consensus sets must be the same.
+    pub validators: Vec<PrivateKey>,
+}
+
+impl ChainInfo {
+    /// Creates Chain info from the standard genesis test suite.
+    ///
+    /// This is useful when you want to test the treasury for the first time.
+    pub fn standard_genesis(chain_name: String) -> Self {
+        let (reserved_state, validators) = test_utils::generate_standard_genesis(4);
+        Self {
+            chain_name,
+            last_finalized_header: reserved_state.genesis_info.header.clone(),
+            last_finalization_proof: reserved_state.genesis_info.genesis_proof.clone(),
+            reserved_state,
+            validators: validators
+                .into_iter()
+                .map(|(_, private_key)| private_key)
+                .collect(),
+        }
+    }
+}
+
+/// A simple transfer scneario.
+///
+/// The treasury contract must be synchronized with the given `chain_info`
+/// and `initial_contract_sequence` as its initial state.
+pub async fn scenario_1(
+    chain_info: ChainInfo,
+    initial_contract_sequence: u128,
+    token_address: HexSerializedVec,
+    temporary_receiver_address: HexSerializedVec,
+    sc: impl SettlementChain,
+) {
+    // Setup the on-chain state
+    let mut csv = CommitSequenceVerifier::new(
+        chain_info.last_finalized_header.clone(),
+        chain_info.reserved_state.clone(),
+    )
+    .unwrap();
+
+    // Query the initial status
+    let intial_balance = sc
+        .get_treasury_fungible_token_balance(token_address.clone())
+        .await
+        .unwrap();
+    let initial_treasury_header = sc.get_light_client_header().await.unwrap();
+    let contract_sequence = sc.get_contract_sequence().await.unwrap();
+    let initial_temporary_receiver_balance = sc
+        .eoa_get_fungible_token_balance(temporary_receiver_address.clone(), token_address.clone())
+        .await
+        .unwrap();
+    assert_eq!(initial_treasury_header, chain_info.last_finalized_header);
+    assert_eq!(contract_sequence, initial_contract_sequence);
+
+    // Apply transactions
+    let mut txes = Vec::new();
```

Comment:
`txes` -> `txns` would be better.

Reply:
Isn't `txes` more common? I saw it multiple times from multiple sources.

Reply:
I think `txns` is more appropriate. For example, google gives me about `taxes` when using `txes`, and `transactions` for `txns`.

Reply:
Ok I think you're right. However I just realized that `transactions` seems the best 😁

Reply:
fixed